OpenRA.Mods.Common/Traits/Conditions/ConditionManager.cs

```diff
@@ -191,6 +222,54 @@ public int GrantCondition(Actor self, string condition, bool external = false, i
 			return token;
 		}
 
+		/// <summary>Replaces granting of a condition by given token with a specified condition.</summary>
+		/// <returns>The token that is used to revoke the replacement condition else given token if failure.</returns>
+		/// <param name="oldToken">Token for revocation.</param>
+		/// <param name="newCondition">Replacement condition to grant.</param>
+		/// <param name="external">Validate against the external condition whitelist.</param>
+		/// <param name="duration">Automatically revoke condition after this delay if non-zero.</param>
+		public int ReplaceCondition(Actor self, int oldToken, string newCondition, bool external = false, int duration = 0)
```

Comment:
I'd prefer that we didn't expose this, it just adds complexit.
If a trait wants to change what it grants it should explicitly revoke the old one and grant the new one.